hedera-node/docs/design/services/smart-contract-service/hbar-allowance-approval.md

```diff
@@ -0,0 +1,97 @@
+
+| Function Selector Hash | Function Signature                                         | Response               |                                                                                                      | 
+|------------------------|------------------------------------------------------------|------------------------|------------------------------------------------------------------------------------------------------|
+| 0xfec46666             | hbarAllowance(address owner, address spender)              | (ResponseCode, int256) | The response code from the call and the amount of hbar allowances currently available to the spender | 
+| 0xa0918464             | hbarApprove(address owner, address spender, int256 amount) | ResponseCode           | The response code from the call                                                                      |
+
+Similar to implementation that exist for `HTS` function calls, the following new classes will be introduced:
+
+- HbarAllowanceTranslator
+- HbarAllowanceCall
+- HbarApproveTranslator
+- HbarApproveCall
+
+### Support for Proxy Contract
+The following code path will be introduced in order to support calls to the `hbarAllowance` and `hbarApprove` functions for EOA accounts via the `IHRC632` interface.
+
+1. During the `MessageFrame` construction for top level calls, the `to` address will be examined to determine if it is an EOA account address. 
+2. If true, the function selector will be checked to determine if it is a call to the `hbarAllowance` or `hbarApprove` functions.
+3. If true, a state will be set via the `ProxyWorldUpdater` to indicate to the `DispatchingEvmFrameState` to provide the bytecode for the `redirectForAccount` proxy contract when queried.
+4. If neither of the above conditions are met, execution will proceed as previously and the `0x` bytecode will be provided to execute.
+
+Internal calls to the `hbarAllowance` and `hbarApprove` functions will only be allowed via the `IHederaAccountService` interface from contracts and thus does not require special handling.
+
+### Hedera Token Service
+
+For `hbarAprrove`, the system must add the approval for the given account or contract to the spender account.  This is achieved via dispatching a request to the Token Service.
+Once the smart contract service dispatches the transaction to the Token Service it performs the following steps:
+
+1. Validate semantic correctness of the transaction.
+2. Grant approval to the spender for the allowance amount from the effective owner.
+
+The implementation can be found in the `CryptoApproveAllowanceHandler` class.
+
+## Acceptance Tests
+In addition to unit tests and xTests, the following acceptance tests will be implemented to verify the functionality and security of the `hbarAllowance` and `hbarApprove` functions.
+
+### BDD Tests
+
+#### Positive Tests
+- Test that an EOA can call the `hbarApprove` function and grant an allowance to another account only if the EOA signs the transaction.
```

Comment:
As a positive test this would be "Test that an EOA can call the `hbarApprove` function and grant an allowance to another account **when** the EOA signs the transaction".  The negative test for this is the first one in the "negative tests" section below.

Similarly for the second positive test here.